data-load/user-load.sh

```diff
@@ -0,0 +1,100 @@
+#!/bin/bash
+#
+# Script to load a batch of new users into OAuth for new service access.
+#
+# Parameters:
+#       1. ENV - [t3|t2|preprod|prod]
+#       2. CLIENT - the client Id and secret, comma-separated
+#       3. USER - the name of the client used to authenticate 
+#       4  BATCH - the size of batches (with 30 second pauses between them)
+#       5. FILE - the name of the file containing the user data
+#
+# Example:
+# 
+# $ ./user_load.sh t3 <client>:<secret> THARRISON_ADM 20 users-data.txt | tee output.txt
+#
+# File format for users:
+#
+#  A comma-separated file containing fields
+#        1) username
+#        2) email address
+#        3) First name
+#        4) Last name
+#        5) group code
+#
+
+ENV=${1?No environment specified}
+CLIENT=${2?No client specified}
+USER=${3?No user specified}
+BATCH=${4?No batch size specified}
+FILE=${5?No file specified}
+
+# Set the environment-specific hostname for the oauth2 service
+if [[ "$ENV" == "t3" ]]; then
+  HOST="https://gateway.t3.nomis-api.hmpps.dsd.io"
+else 
+  HOST="https://gateway.$ENV.nomis-api.service.hmpps.dsd.io"
+fi
+
+# Check whether the file exists and is readable
+if [[ ! -f "$FILE" ]]; then
+  echo "Unable to find file $FILE"
+  exit 1
+fi
+
+# Get  token for the client name / secret and store it in a hidden file .access_token
```

Comment:
doesn't store in a hidden fie, we just use `jq` to extract out the `access_token` from the json

Reply:
Corrected to stored in env var TOKEN.